### PR TITLE
Reactify charts

### PR DESCRIPTION
**Pull request by Velenir**
Some suggestions to make charts more react-like

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr819--dexreact.review.gnosisdev.com

File: src/components/OrderBookWidget.tsx [hunk 1]
```diff
@@ -157,18 +162,19 @@ const draw = (baseToken: TokenDetails, quoteToken: TokenDetails, dataSource: str
...
   useEffect(() => {
-    if (chart) {
-      chart.dispose()
-    }
-    setChart(draw(baseToken, quoteToken, orderbookUrl(baseToken, quoteToken, networkId)))
+    if (!mountPoint.current) return
+
+    const chart = draw(mountPoint.current, baseToken, quoteToken, orderbookUrl(baseToken, quoteToken, networkId))
+
+    return (): void => chart.dispose()
```
Comment (fleupold) [inline src/components/OrderBookWidget.tsx:173]: I think we still need to dispose on every redraw (e.g. when tokens change)

Comment (Velenir) [reply src/components/OrderBookWidget.tsx:173]: And that's exactly how  `useEffect` works
```ts
useEffect(() => {
...
    return (): void => chart.dispose() //runs on clean-up stage
  }, [baseToken, quoteToken, networkId]) //deps
```
deps change -> cleanup function runs -> actual function runs -> deps change -> clean-up ...

**Comment by Velenir** [comment]
> Awesome, thanks a lot! Can I just merge into my branch?

Go for it :+1:

**Pull request merged**